### PR TITLE
ComponentModel: Add method to export inner interfaces

### PR DESCRIPTION
**Pull request by ErikMichelson**

### CODE REVIEW

File: src/ComponentModel.cpp [hunk 1]
```diff
@@ -1357,6 +1357,56 @@ std::vector<ComponentModelPtr> xtypes::ComponentModel::import_from_basic_model(c
...
+    XTypeRegistryPtr reg = registry.lock();
+    if (!reg)
+    {
+        throw std::invalid_argument("ComponentModel::export_inner_interface: no registry");
+    }
+    const auto new_name = parent->get_name() + ":" + inner_interface_name;
+    std::cout << "Creating new interface " << new_name << " for " << this->get_name() << "\n";
+    
+    const InterfaceModelPtr model(inner_interface->get_type());
+    InterfacePtr interface = std::static_pointer_cast<Interface>(reg->instantiate<Interface>());
```
Comment (m0rsch1) [inline src/ComponentModel.cpp]: You could reuse outer_interface here

File: src/ComponentModel.cpp [hunk 2]
```diff
@@ -1357,6 +1357,56 @@ std::vector<ComponentModelPtr> xtypes::ComponentModel::import_from_basic_model(c
...
+    interface->set_all_unknown_facts_empty();
```
Comment (m0rsch1) [inline src/ComponentModel.cpp]: Either make this optional (see ComponentModel::instantiate with_empty_facts) or leave this out.
If all the unknown facts are made empty they could potentially override existing facts in the database.

**Review by m0rsch1 — CHANGES_REQUESTED** [review]

File: templates/component_model.yaml [hunk 3]
```diff
@@ -221,6 +221,16 @@ methods:
       type: MAP(XTYPE(InterfacePtr), VECTOR(XTYPE(InterfacePtr)))
     description: "This functions resolves any interfaces of inner parts which do not match any of the parts' model interfaces and a list of possible future matches."
 
+  export_inner_interface:
+    arguments:
+      - name: inner_interface
+        type: XTYPE(Interface)
+      - name: with_empty_facts
```
Comment (m0rsch1) [inline templates/component_model.yaml:228]: Assign a default value here

**Review by m0rsch1 — CHANGES_REQUESTED** [review]

File: templates/component_model.yaml (hunk 3 above)
Comment (ErikMichelson) [reply templates/component_model.yaml:228]: Based on the previous comment, it should probably be `false` as default?

**Review by m0rsch1 — APPROVED** [review]
Looks good to me

**Pull request merged**